bindings/pyroot/pythonizations/python/ROOT/_pythonization/_roofit/_roodatahist.py

```diff
@@ -50,3 +50,190 @@ def plotOn(self, *args, **kwargs):
         # The keywords must correspond to the CmdArg of the `plotOn` function.
         args, kwargs = _kwargs_to_roocmdargs(*args, **kwargs)
         return self._plotOn(*args, **kwargs)
+
+    @property
+    def shape(self):
+        import ROOT
+        import numpy as np
+
+        shape = []
+
+        var_is_category = self._var_is_category()
+
+        for i_var, var in enumerate(self.get()):
+            shape.append(var.numBins())
+
+        def get_len_from_shape():
+            n_size_check = 1
+            for n in shape:
+                n_size_check = n_size_check * n
+            return n_size_check
+
+        assert get_len_from_shape() == len(self)
+
+        return tuple(shape)
+
+    def __len__(self):
+        return self.numEntries()
+
+    def _to_array(self, buffer):
+        import numpy as np
+
+        # check if buffer is nullptr (for some reason comparing with ROOT.nullptr doesn't work)
```

Comment:
What's the type of buffer here?

Reply:
Perhaps adding some docs would be good too.

Reply:
The function is meant to create a numpy array from a `double *`. The type of buffer usually a `cppyy.LowLevelView`, which is what you get when you call a C++ function that returns a `double *`. I'll write some doc!

Reply:
Ok thanks, my question was more in the direction: if a `cppyy.LowLevelView` points null, it's not equal to `ROOT.nullptr`?

Reply:
I just checked that, and indeed:
```
>>> ROOT.gInterpreter.Declare("double * foo() { return nullptr; }")
True
>>> buffer = ROOT.foo()
>>> buffer == ROOT.nullptr
False
```
but what you can do is use the boolean value of the `cppyy.LowLevelView`:
```
if not buffer: return None
...
```

Reply:
Ah! Nice, I will to de boolean check then, as it's much cleaner :+1: